calico-enterprise_versioned_docs/version-3.18/reference/resources/calico-labels.mdx

```diff
@@ -0,0 +1,84 @@
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+name: foo-and-bar
+spec:
+namespaceSelector: projectcalico.org/name in {"foo", "bar"}
+selector: color == "red"
+types:
+  - Ingress
+ingress:
+  - action: Allow
+source:
+namespaceSelector: projectcalico.org/name == "baz"
+destination:
+ports:
+  - 8080
+```
+
+:::note
+Kubernetes also has [automatic labeling](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#automatic-labelling), for example: `kubernetes.io/metadata.name`.
+:::
+
+## ServiceAccount
+
+Similarly, the `projectcalico.org/name` label is applied to ServiceAccounts and allows for matching by name in a `serviceAccountSelector`.
+
+## Use the correct selector with labels in policies
+
+Calico labels must be used with the correct selector or the policy will not work as designed (and there are no error messages in Manager UI or when applying the YAML).
+
+| Calico label         | Usage requirements                                              | Use in these resources...           |
+| --------------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| `projectcalico.org/name`                                     | Use with a **namespaceSelector** or **serviceAccountSelector**. | - Network policy<br />- Staged network policy<br /><br />Namespaced resources that apply only to workload endpoint resources in the namespace.<br /> |
+| `projectcalico.org/namespace`                                | Use the label as the label name, and a namespace name as the value to compare against (for example projectcalico.org/namespace == "default"). | - Global network policy<br />- Staged global network policy<br /><br />Cluster-wide (non-namespaced) resources that apply to workload endpoint resources in all namespaces, and to host endpoint resources. |
+
+## Workload endpoints
+
+Workload endpoints (which represent Pods in Kubernetes, or VM instances in OpenStack) receive several automatic labels:
+
+* `projectcalico.org/orchestrator` is applied to all workload endpoints and allows Kubernetes Pods to be distinguished from OpenStack VM instances, and from HostEndpoints (which do not have the label):
+
+* `has(projectcalico.org/orchestrator)` matches only workload endpoints
+* `projectcalico.org/orchestrator == "k8s"` matches only Kubernetes Pods
+
+* For workload endpoints that represent Kubernetes Pods, `projectcalico.org/namespace` contains the name of the Pod's namespace. `projectcalico.org/namespace` predates the addition of `namespaceSelector` fields to GlobalNetowrkPolicies; it serves the same purpose as the `projectcalico.org/name` label in a `namespaceSelector` field.  The following GlobalNetworkPolicy is exactly equivalent to the one above:
```

Comment:
I _expect_ that `projectcalico.org/namespace` also applies to NetowrkSets, but I'm not 100% sure.  NetowrkSets get convered similarly to WEPs.

Reply:
I don't see anything documented, but that's not to say...